src/components/ScreenWrapper/index.js

```diff
@@ -1,155 +1,156 @@
 import {Keyboard, View, PanResponder} from 'react-native';
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import _ from 'underscore';
 import lodashGet from 'lodash/get';
 import {PickerAvoidingView} from 'react-native-picker-select';
+import {useNavigation} from '@react-navigation/native';
 import KeyboardAvoidingView from '../KeyboardAvoidingView';
 import CONST from '../../CONST';
 import styles from '../../styles/styles';
 import HeaderGap from '../HeaderGap';
 import OfflineIndicator from '../OfflineIndicator';
-import compose from '../../libs/compose';
-import withNavigation from '../withNavigation';
-import {withNetwork} from '../OnyxProvider';
 import {propTypes, defaultProps} from './propTypes';
 import SafeAreaConsumer from '../SafeAreaConsumer';
 import TestToolsModal from '../TestToolsModal';
-import withKeyboardState from '../withKeyboardState';
-import withWindowDimensions from '../withWindowDimensions';
-import withEnvironment from '../withEnvironment';
 import toggleTestToolsModal from '../../libs/actions/TestTool';
 import CustomDevMenu from '../CustomDevMenu';
 import * as Browser from '../../libs/Browser';
+import useWindowDimensions from '../../hooks/useWindowDimensions';
+import useKeyboardState from '../../hooks/useKeyboardState';
+import useEnvironment from '../../hooks/useEnvironment';
+import useNetwork from '../../hooks/useNetwork';
 
-class ScreenWrapper extends React.Component {
-    constructor(props) {
-        super(props);
+function ScreenWrapper({
+    shouldEnableMaxHeight,
+    includePaddingTop,
+    keyboardAvoidingViewBehavior,
+    includeSafeAreaPaddingBottom,
+    shouldEnableKeyboardAvoidingView,
+    shouldEnablePickerAvoiding,
+    headerGapStyles,
+    children,
+    shouldShowOfflineIndicator,
+    offlineIndicatorStyle,
+    style,
+    shouldDismissKeyboardBeforeClose,
+    onEntryTransitionEnd,
+}) {
+    const {windowHeight, isSmallScreenWidth} = useWindowDimensions();
+    const keyboardState = useKeyboardState();
+    const {isDevelopment} = useEnvironment();
+    const {isOffline} = useNetwork();
+    const navigation = useNavigation();
+    const [didScreenTransitionEnd, setDidScreenTransitionEnd] = useState(false);
+    const maxHeight = shouldEnableMaxHeight ? windowHeight : undefined;
+    const isKeyboardShown = lodashGet(keyboardState, 'isKeyboardShown', false);
 
-        this.panResponder = PanResponder.create({
-            onStartShouldSetPanResponderCapture: (e, gestureState) => gestureState.numberActiveTouches === CONST.TEST_TOOL.NUMBER_OF_TAPS,
-            onPanResponderRelease: toggleTestToolsModal,
-        });
+    const panResponder = PanResponder.create({
```

Comment:
Looks like PanResponder should be memorized somehow. By the official [doc](https://reactnative.dev/docs/panresponder), they use `useRef`.